bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -736,9 +727,8 @@ ClientRequestMsg *ReplicaImp::addRequestToPrePrepareMessage(ClientRequestMsg *&n
   }
   primaryCombinedReqSize -= nextRequest->size();
   requestsQueueOfPrimary.pop();
-  delete nextRequest;
```

Comment:
what is the change which cause you not to deleted the pointer?  pointeer is still raw, right?

Reply:
The raw pointer is passed to the function, but it comes from std::queue<std::unique_ptr<ClientRequestMsg>> requestsQueueOfPrimary;. Note the smart pointer here. So upon pop() call it is deleted.